### PR TITLE
Fix issue with URI type in the enum

### PR DESCRIPTION
**Pull request by hhomar**
This is a followup to #16929 to fix the compilation errors with the java ok-http-gson. I also tested compilation against all java libraries.

The modelEnum template needed an additional update to CodegenModel to check if the format is set as a URI as the isUri check was returning false. I followed the same code as isUuid, but noticed a comment in DefaultCodeGen about UUID being a rare use case and may be removed in future [DefaultCodegen.java#L3031](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L3031), so I don't know if this change is the best approach.

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)

### CODE REVIEW

**Comment by wing328** [comment]
thanks for the PR

cc @OpenAPITools/generator-core-team

**Comment by wing328** [comment]
can you please resolve the merge conflicts when you've time?

**Comment by hhomar** [comment]
Conflict resolved.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenModel.java [hunk 1]
```diff
@@ -177,6 +177,7 @@ public class CodegenModel implements IJsonSchemaValidationProperties {
     private boolean hasDiscriminatorWithNonEmptyMapping;
     private boolean isAnyType;
     private boolean isUuid;
+    private boolean isUri;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenModel.java:180]: FYI @OpenAPITools/generator-core-team

**Comment by wing328** [comment]
tested locally to confirm the fix. thanks for the PR

**Pull request merged**